### ISSUE TITLE
Adversarial training is meaningless, maybe the codes are wrong?

### ISSUE THREAD

**Issue by machengcheng2016** [VITA-Group/Aug-NeRF#5]
Hello,
Really nice job! But I have some concerns with the released codes.

According to your codes, [`AdvNeRFNet`](https://github.com/VITA-Group/Aug-NeRF/blob/main/run_nerf.py#L227) has three components, that are [`AdvNeRFNet.nerf_adv`](https://github.com/VITA-Group/Aug-NeRF/blob/main/models/adv_nerf_net.py#L34), [`AdvNeRFNet.nerf`](https://github.com/VITA-Group/Aug-NeRF/blob/main/models/nerf_net.py#L49) and [`AdvNeRFNet.nerf_fine`](https://github.com/VITA-Group/Aug-NeRF/blob/main/models/nerf_net.py#L54). 

During training, the [`loss_adv`](https://github.com/VITA-Group/Aug-NeRF/blob/main/engines/trainer_adv.py#L141) and [`adv_perturb`](https://github.com/VITA-Group/Aug-NeRF/blob/main/engines/trainer_adv.py#L123) are only related with the update of `AdvNeRFNet.nerf_adv`, and the [`loss_clean`](https://github.com/VITA-Group/Aug-NeRF/blob/main/engines/trainer_adv.py#L141) is only related with the update of `AdvNeRFNet.nerf` and `AdvNeRFNet.nerf_fine`. 

During inference, only `AdvNeRFNet.nerf` and `AdvNeRFNet.nerf_fine` are involved.

After carefully checking your codes, I find that `AdvNeRFNet.nerf_adv`, `AdvNeRFNet.nerf` and `AdvNeRFNet.nerf_fine` are independent with each other and there are no common components before or after them, so the existence of the `AdvNeRFNet.nerf_adv` branch seems meaningless to the training and inference step of conventional NeRF.

What do you think? Is there any details I ignore?

**Comment by machengcheng2016** [comment]
I am a huge fan of adversarial training, and I have also checked your previous work [CV_A-FAN](https://github.com/VITA-Group/CV_A-FAN). 
In the classification task of CV_A-FAN, there seems no additional "adversarial" branch within the ResNet. So the ResNet weights can be adversarially trained upon benign examples and adversarial examples.
Maybe there should also exist no additional branch in Aug-NeRF?

**Comment by peihaowang** [comment]
Hi,
Thanks for raising this issue. I have checked the code, and I guess the current version may indeed have some problems. In our original implementation, I remember the adversarial attack should be added through one of the branches in NeRF (recall that in NeRF, the MLP has two branches, one for density prediction, and another for color prediction). One quick solution might be replacing all the `nerf_adv` with `nerf_fine`/`nerf`. I'll be working on correcting it. Thanks for your patience.

**Comment by machengcheng2016** [comment]
Yeah yeah, I think `nerf_adv` should be replaced with `nerf_fine`/`nerf` as well.
